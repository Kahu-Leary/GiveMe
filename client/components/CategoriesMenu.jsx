import React, { useState } from 'react'

function CategoriesMenu () {
  const [isActive, setActive] = useState(false)

  const handleToggle = () => {
    switch (isActive) {
      case false: setActive(true)
        break
      case true: setActive(false)
        break
    }
  }

  return (
    <div >
      <div className={isActive ? 'dropdown is-active' : 'dropdown'}>
        <div className="dropdown-trigger">
          <button onClick={handleToggle} className="button primaryBtn " aria-haspopup="true" aria-controls="dropdown-menu3">
            Categories
          </button>
        </div>
        <div className="dropdown-menu" id="dropdown-menu3" role="menu">
          <div className="dropdown-content">

            <a href="#" className="dropdown-item">Books</a>

            <a href="#" className="dropdown-item">Furniture</a>

            <a href="#" className="dropdown-item">Toys</a>

          </div>
        </div>
      </div>
    </div>
  )
}

export default CategoriesMenu

// {items.items.map((item) => {
//   return <li className="imageGridItem" key={item.id} >
//     {item.category}
//   </li>
