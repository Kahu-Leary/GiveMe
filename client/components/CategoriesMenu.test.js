import React from 'react'

import renderer from 'react-test-renderer'

import CategoriesMenu from './CategoriesMenu'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <CategoriesMenu/>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
