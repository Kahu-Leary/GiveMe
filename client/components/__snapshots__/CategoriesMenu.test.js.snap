// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="dropdown"
  >
    <div
      className="dropdown-trigger"
    >
      <button
        aria-controls="dropdown-menu3"
        aria-haspopup="true"
        className="button primaryBtn "
        onClick={[Function]}
      >
        Categories
      </button>
    </div>
    <div
      className="dropdown-menu"
      id="dropdown-menu3"
      role="menu"
    >
      <div
        className="dropdown-content"
      >
        <a
          className="dropdown-item"
          href="#"
        >
          Books
        </a>
        <a
          className="dropdown-item"
          href="#"
        >
          Furniture
        </a>
        <a
          className="dropdown-item"
          href="#"
        >
          Toys
        </a>
      </div>
    </div>
  </div>
</div>
`;
